PorticoExportDom.inc.php

```diff
@@ -140,41 +140,43 @@ function &generateArticleDom(&$doc, &$journal, &$issue, &$section, &$article) {
 		}
 
 		// supplementary file links
-		foreach ($article->getSuppFiles() as $suppFile) {
+		$galleyDao =& DAORegistry::getDAO('ArticleGalleyDAO');
+		$genreDao =& DAORegistry::getDAO('GenreDAO');
+		$suppFiles= $galleyDao->getBySubmissionId($article->getId(), $article->getContextId())->toArray();
+		
+		foreach ($suppFiles as $suppFile) {
 			$supplementaryMaterialNode =& XMLCustomWriter::createChildWithText($doc, $articleMetaNode, 'supplementary-material');
-			XMLCustomWriter::setAttribute($supplementaryMaterialNode, 'xlink:href', $suppFile->getFileName());
+			XMLCustomWriter::setAttribute($supplementaryMaterialNode, 'xlink:href', $suppFile->getLocalizedName());
 			XMLCustomWriter::setAttribute($supplementaryMaterialNode, 'content-type', $suppFile->getFileType());
-			}
+		}
 
 		// galley links
-		import('classes.file.ArticleFileManager');
-		$articleFileManager = new ArticleFileManager($article->getId());
-		foreach ($article->getGalleys() as $galley) {
-			$selfUriNode =& XMLCustomWriter::createChildWithText($doc, $articleMetaNode, 'self-uri', $galley->getFileName());
-			XMLCustomWriter::setAttribute($selfUriNode, 'xlink:href', $galley->getFileName());
+		import('lib.pkp.classes.file.SubmissionFileManager');
+		$submissionFileManager = new SubmissionFileManager($article->getContextId(),$article->getId());
+			foreach ($galleys as $galley) {
+			$selfUriNode =& XMLCustomWriter::createChildWithText($doc, $articleMetaNode, 'self-uri', $galley->getName($galley->getLocale()));
+			XMLCustomWriter::setAttribute($selfUriNode, 'xlink:href', $galley->getName($galley->getLocale()));
 			XMLCustomWriter::setAttribute($selfUriNode, 'content-type', $galley->getFileType());
 			
-			// check for HTML-ness, and if there are additional galley files include them 
-			$isHtml = $galley->isHTMLGalley();
-	
-			if ($isHtml) {
-				$styleFile = $galley->getStyleFile();
-				if ($styleFile) {
-					$styleNode =& XMLCustomWriter::createChildWithText($doc, $articleMetaNode, 'related-object');
-					XMLCustomWriter::setAttribute($styleNode, 'xlink:href', $styleFile->getFileName());
-					XMLCustomWriter::setAttribute($styleNode, 'content-type', $styleFile->getFileType());
-					$objectNode =& XMLCustomWriter::createChildWithText($doc, $styleNode, 'object-id', $styleFile->getFileName());
-				}
+			// check for HTML-ness, and if there are additional galley files include them **FIGURE OUT WHAT THE REPLACEMENT METHODS FOR THIS ARE**
```

Comment:
There's a note here that probably needs to be deleted. If it's still a question that needs answering, you can ask either myself, Michael or Alec.